libraries/node-core-library/src/Terminal/PrefixProxyTerminalProvider.ts

```diff
@@ -58,22 +93,35 @@ export class PrefixProxyTerminalProvider implements ITerminalProvider {
     let currentIndex: number = 0;
     // eslint-disable-next-line @rushstack/no-new-null
     let newlineMatch: RegExpExecArray | null;
+    let isFirstLoop: boolean = true;
+    const startedOnNewLine: boolean = this._isOnNewline;
+    let currentPrefix: string = this._isOnNewline ? this._getPrefix() : '';
+
     while ((newlineMatch = this._newlineRegex.exec(data))) {
+      if (!isFirstLoop && !startedOnNewLine) {
+        currentPrefix = this._getPrefix();
```

Comment:
I wonder if maintaining the same prefix across multiple lines of a string containing newlines is what's expected? `getPrefix` seems to imply that it'd be called each time a prefix is required, which is not what's happening here.